cmd/prometheus/main.go

```diff
@@ -160,9 +160,15 @@ func main() {
 		"Maximum duration compacted blocks may span. For use in testing. (Defaults to 10% of the retention period).").
 		Hidden().PlaceHolder("<duration>").SetValue(&cfg.tsdb.MaxBlockDuration)
 
-	a.Flag("storage.tsdb.retention", "How long to retain samples in storage.").
+	a.Flag("storage.tsdb.retention", "How long to retain samples in storage. (This flag has been deprecated. Use storage.tsdb.retention.time as a replacement.)").
 		Default("15d").SetValue(&cfg.tsdb.Retention)
 
+	a.Flag("storage.tsdb.retention.time", "How long to retain samples in storage.").
+		Default("15d").SetValue(&cfg.tsdb.Retention)
+
+	a.Flag("storage.tsdb.retention.size", "Maximum number of bytes that can be stored for blocks.").
```

Comment:
as @gouthamve 	mentioned on irc lets mark this experimental to warn people and reduce bugfix urgency. again using some midleware pattern to show a warning when starting Prometheus with this flag and not just a comment in the code.